src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java

```diff
@@ -559,6 +559,17 @@ private void writeDependencyXML(Set<Artifact> artifacts) {
                 writer.startElement("artifactId");
                 writer.writeText(artifact.getArtifactId());
                 writer.endElement();
+                String key = String.format(
+                        "%s:%s:%s:%s",
+                        artifact.getGroupId(),
+                        artifact.getArtifactId(),
+                        artifact.getType(),
+                        (artifact.getClassifier() != null ? ":" : artifact.getClassifier()));
```

Comment:
is this conditional correct? what does `key` look like when `classifier` is either null or not null?

Reply:
Oh no i thought i committed my fix for that! It's supposed to return `: + the classifier` if it's not null, and an empty string if it is